eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -0,0 +1,227 @@
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.Resource;
+import uk.ac.ebi.eva.commons.core.models.Annotation;
+import uk.ac.ebi.eva.commons.core.models.AnnotationMetadata;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantSourceEntryWithSampleNames;
+import uk.ac.ebi.eva.commons.core.models.ws.VariantWithSamplesAndAnnotation;
+import uk.ac.ebi.eva.commons.mongodb.services.AnnotationMetadataNotFoundException;
+import uk.ac.ebi.eva.commons.mongodb.services.VariantWithSamplesAndAnnotationsService;
+import uk.ac.ebi.eva.lib.eva_utils.DBAdaptorConnector;
+import uk.ac.ebi.eva.lib.eva_utils.MultiMongoDbFactory;
+import uk.ac.ebi.eva.lib.utils.QueryResponse;
+import uk.ac.ebi.eva.lib.utils.QueryResult;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@RestController
+@RequestMapping(value = "/v2/variants", produces = "application/json")
+@Api(tags = {"variants"})
+public class VariantWSServerV2 extends EvaWSServer {
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @GetMapping(value = "/{variantId}")
+    public Resource<QueryResponse> getCoreInfo(@PathVariable("variantId") String variantId,
+                                               @RequestParam(name = "species") String species,
+                                               @RequestParam(name = "annot-vep-version", required = false)
+                                                       String annotationVepVersion,
+                                               @RequestParam(name = "annot-vep-cache-version", required = false) String
+                                                       annotationVepCacheVersion,
+                                               HttpServletResponse response) {
+        initializeQuery();
+
+        String errorMessage = checkErrorHelper(variantId, annotationVepVersion, annotationVepCacheVersion, species);
+        if (errorMessage != null) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setErrorQueryResponse(errorMessage));
+        }
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
+
+        List<VariantWithSamplesAndAnnotation> variantEntities;
+        Long numTotalResults;
+
+        try {
+            variantEntities = getVariantEntitiesByParams(variantId, annotationVepVersion, annotationVepCacheVersion);
+            numTotalResults = (long) variantEntities.size();
+        } catch (AnnotationMetadataNotFoundException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setQueryResponse(ex.getMessage()));
+        }
+
+        List<VariantWithSamplesAndAnnotation> rootVariantEntities = new ArrayList<>();
```

Comment:
Can we build a simpler object than this, given that this endpoint is not going to report samples nor annotation?

Reply:
I have created a new class named VariantCoreInfo. Would that be okay?

Reply:
Check if uk.ac.ebi.eva.commons.core.models.pipeline.Variant works for this before creating another bean

Reply:
I tried using this now, but the `Type` and `length` field doesn't get retrieved along with the response as there is no datamember of the same in `uk.ac.ebi.eva.commons.core.models.pipeline.Variant` class

Reply:
The problem comes from [this section of the code](https://github.com/EBIvariation/eva-ws/blob/master/eva-server/src/main/java/uk/ac/ebi/eva/server/configuration/JacksonConfiguration.java#L50-L54), which only searches for attributes, not getter methods. We had to do this for the REST API v1 because the model classes are complex, and they have a lot of getters that provide different views to the same information. But the `Variant` class should not give any problems to be retrieved via getter methods.

So we have 2 choices:

1. Create 2 different `ObjectMapper`s and configure them based on the URL (`/v1` for fields and `/v2` for getters). I don't know how we can do this but I guess it is possible.
2. Create a new Maven module that only contains the API v2, making the `ObjectMapper` configuration straightforward.

Reply:
I will check this out and will get back to you :)

Reply:
Fixed by using `objectMapper.addMixIn(Variant.class, VariantMixin.class)` :+1: